samples/modules/cmsis_dsp/moving_average/src/main.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024 Benjamin Cabé <benjamin@zephyrproject.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+
+#include "arm_math.h"
+
+#define NUM_TAPS   10 /* Number of taps in the FIR filter (length of the moving average window) */
+#define BLOCK_SIZE 32 /* Number of samples processed per block */
+
+/*
+ * Filter coefficients are all equal for a moving average filter. Here, 1/NUM_TAPS = 0.1f.
+ */
+q31_t firCoeffs[NUM_TAPS] = {0x0CCCCCCD, 0x0CCCCCCD, 0x0CCCCCCD, 0x0CCCCCCD, 0x0CCCCCCD,
+			     0x0CCCCCCD, 0x0CCCCCCD, 0x0CCCCCCD, 0x0CCCCCCD, 0x0CCCCCCD};
+
+arm_fir_instance_q31 sFIR;
+q31_t firState[NUM_TAPS + BLOCK_SIZE - 1];
+
+int main(void)
+{
+	q31_t input[BLOCK_SIZE];
+	q31_t output[BLOCK_SIZE];
+	uint32_t start, end;
+
+	/* Initialize input data with a ramp from 0 to 31 */
+	for (int i = 0; i < BLOCK_SIZE; i++) {
+		input[i] = i << 24; /* Convert to Q31 format */
+	}
+
+	/* Initialize the FIR filter */
+	arm_fir_init_q31(&sFIR, NUM_TAPS, firCoeffs, firState, BLOCK_SIZE);
+
+	/* Apply the FIR filter to the input data and measure how many cycles this takes */
+	start = k_cycle_get_32();
+	arm_fir_q31(&sFIR, input, output, BLOCK_SIZE);
+	end = k_cycle_get_32();
```

Comment:
Does averaging 32 samples take long enough to show up at the cycle resolution?
The standard API for profiling durations is `zephyr/timing/timing.h` with `CONFIG_TIMING_FUNCTIONS=y` from my understanding.